File: WebAPI/BrewMapAPI/BrewMapAPI/Models/Review.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace BrewMapAPI.Models
{
    public class Review
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("userId")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string UserId { get; set; }

        [BsonElement("target")]
        public ReviewTarget Target { get; set; }

        [BsonElement("rating")]
        public int Rating { get; set; } // 1-5

        [BsonElement("comment")]
        public string? Comment { get; set; }

        [BsonElement("isVisible")]
        public bool IsVisible { get; set; } = true;

        [BsonElement("reportCount")]
        public int ReportCount { get; set; } = 0;

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }

    public class ReviewTarget
    {
        [BsonElement("type")]
        public string Type { get; set; } // E.g. "location" or "product"

        [BsonElement("id")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
    }
}

File: WebAPI/MobileApp/app/src/main/java/hr/algebra/mobileapp/models/review/Review.kt
package hr.algebra.mobileapp.models.review

/**
 * Mirrors `ReadReview` DTO from the BrewMap API.
 *
 * [targetType] is either `"locationService"` or `"product"` (drinkService).
 * [targetId]   is the MongoDB ObjectId of the reviewed locationService or drinkService.
 * [rating]     is an integer 1–5.
 */
data class Review(
    val id: String,
    val userId: String,
    val username: String?,
    val targetType: String,
    val targetId: String,
    val rating: Int,
    val comment: String?,
    val isVisible: Boolean,
    val reportCount: Int,
    val createdAt: String,
    val updatedAt: String
)
